llvm/lib/Target/SPIRV/SPIRVPreLegalizerCombiner.cpp

```diff
@@ -147,8 +147,8 @@ bool matchSelectToFaceForward(MachineInstr &MI, MachineRegisterInfo &MRI) {
 
   // Check if FCMP is a comparison between a dot product and 0.
   MachineInstr *DotInstr = MRI.getVRegDef(DotReg);
-  if ((DotInstr->getOpcode() != TargetOpcode::G_INTRINSIC ||
-       cast<GIntrinsic>(DotInstr)->getIntrinsicID() != Intrinsic::spv_fdot)) {
+  if (DotInstr->getOpcode() != TargetOpcode::G_INTRINSIC ||
+      cast<GIntrinsic>(DotInstr)->getIntrinsicID() != Intrinsic::spv_fdot) {
```

Comment:
If we don't have a matcher for intrinsic that takes the Intrinsic ID as an argument, there should be one